app-typescript/components/Alert.tsx

```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
+import classNames from 'classnames';
+
+interface IProps {
+    type?: 'primary' | 'success' | 'warning' | 'alert' | 'highlight' | 'sd-green';
+    style?: 'filled' | 'hollow'; // defaults to 'filled'
+    size?: 'normal' | 'small'; // defaults to 'normal'
+    icon?: 'info' | 'help'; // defaults to 'info'
+}
+
+interface IState {
+    open: boolean;
+}
+
+export class Alert extends React.Component<IProps, IState> {
+    constructor(props: IProps) {
+        super(props);
+        this.state = {
+            open: true,
+        };
+        this.onClose = this.onClose.bind(this);
+    }
+
+    onClose() {
+        this.setState((state) => ({
+            open: !state.open,
```

Comment:
Leave icon as icon, because initial state of alert can also be closed, so restore-icon doesn't have sense. Toggle is not enough for knowing that alert can be opened and closed?

Reply:
> Toggle is not enough for knowing that alert can be opened and closed?

`onToggle` would be a part of the implementation. When you are trying to use a component from client-core and jump to type definitions, you don't see the implementation - only types. And that's a good thing. Types and comments on types should be enough to understand how to use the component.

I'd prefer to call it `restore-icon` because it's only displayed when the alert is hidden which is not obvious otherwise. For example, when I was initially testing the component, I didn't know it can be restored after closing and was thinking that the icon doesn't work - because I expected the icon to be visible somewhere next to the alert text.